src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs

```diff
@@ -96,41 +97,50 @@ private partial class SuggestedActionsSource : IAsyncSuggestedActionsSource
                     // items should be pushed higher up, and less important items shouldn't take up that much space.
                     var currentActionCount = 0;
 
-                    using var _2 = ArrayBuilder<SuggestedActionSet>.GetInstance(out var lowPrioritySets);
+                    var pendingActionSets = new MultiDictionary<CodeActionRequestPriority, SuggestedActionSet>();
 
                     // Collectors are in priority order.  So just walk them from highest to lowest.
                     foreach (var collector in collectors)
                     {
-                        var priority = TryGetPriority(collector.Priority);
-
-                        if (priority != null)
+                        if (TryGetPriority(collector.Priority) is CodeActionRequestPriority priority)
                         {
+                            // See if there are any items we found previously while calling into higher pri providers
+                            // that returned lower pri items. For example 'add import' is a high pri provider, but can
+                            // produce 'low pri' items (like those that add metadata-references). These can be added
+                            // now.
+                            foreach (var set in pendingActionSets[priority])
```

Comment:
I had tried a similar change in my current open PR and noticed an observable delay in these pending items showing up when these are drained at top of the loop for matching priority as compared to draining towards the bottom of this loop for the previous higher priority class. Though my PR also had other changes. Definitely test this out in RoslynDev hive to see any perf impact for items we know will be added to pending set

Reply:
i'll try... though that seems... impossible?  in the past we would have a high pri provider, returning a high or low pri item (there are none that return normal pri items).  The high pri items would show right away, and the low pri items would show at teh end of hte normal group anways (the expensive one).

Now, we still have that a high pri provider puts into high pri or low pri bucket.  The high pri shows right away, and the low pri shows right after the normal pri group finishes (and  should be fast since it's draining prior pending items and we don't actually have any low-pri providers to actually do work).  So it would still be just as long as the normal pri group.

We also have that the normal pri group will pend to the low pri group.  But again taht won't change the speed of the normal pri group, and the low pri group will finish near immediately since it doesn't actually have to do any computation work (no providers match it) and can just add its pending items right away.

Reply:
ok.  tried in roslyn itself, and things feel fine.  Also... all the perf changes that have gone in are great.  Things are so much faster with lightbulbs across teh board!

Reply:
I recall the problem now. In the current approach (drain at the bottom of the loop for prior priority class), these pending action sets show up much earlier compared to the approach you chose here when there are expensive analyzers in the matching priority class. Let me elaborate with an example.

We compute code actions for `High` priority class, and the reported code action sets contains some sets with `Low` prior priority, so we add them to pendingSets. In the approach before your PR, these actions are added at the end of the `Normal` priority sets, and as soon as below code executes, they are added to the light bulb: https://github.com/dotnet/roslyn/blob/3a7a7407ea3c831630ecf2754092c33df3a6e452/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs#L142-L146

However, with the change you have made here, even one slow `Low` priority analyzer will force these already computed `Low` priority pendingSets to show up with that delay. This is because we mark the `Low` priority collector complete only after all `Low` priority code actions have been computed and added to the set. I noticed a big observable perf hit on large files when I moved SymbolStart/SemanticModel analyzers to low priority bucket, and this started causing the code actions in pendingSets with `Low` priority to show up after these analyzers completed, not before, and that is undesirable.

Reply:
ah.  that makes sense.  teh computed pending group should be added at the end of the priority group one higher than the group it is reported for.  That way it is still deprioritized in the list, but shows up immediatley before it's pri class runs, rather than waiting for that pri class to finish.  I can make this change.

Reply:
Ok.  Did this.  @mavasani ptal.